### PR TITLE
(#5440) - fix build and tests on Windows

### PR DESCRIPTION
**Pull request by nolanlawson**
After this fix, you can actually build, test, and develop PouchDB on Windows. I also added some more instructions to `CONTRIBUTING.md` to help out new Windows contributors.

### CODE REVIEW

File: tests/integration/test.prefix.js [hunk 1]
```diff
@@ -38,7 +38,9 @@ describe('test.prefix.js', function () {
 if (typeof process !== 'undefined' &&
     !process.env.LEVEL_ADAPTER &&
     !process.env.LEVEL_PREFIX &&
-    !process.env.ADAPTER) {
+    !process.env.ADAPTER &&
+    // fails on windows with EBUSY - "resource busy or locked", not worth fixing
```
Comment (dharders) [inline tests/integration/test.prefix.js]: Hey @nolanlawson quick heads up... were you using VS Code here when you received the dreaded ` EBUSY - "resource busy or locked"`?

I've experience this (and similar) issues with Windows / Node dev when using VS Code. I found (with a handy little utility _Process Explorer_) that VS Code was the culprit. When I close VS Code and run my scripts, everything works.

YMMV ;)

Comment (nolanlawson) [inline tests/integration/test.prefix.js]: Ah yeah I was running WebStorm at the same time. That may be the issue.

In any case, this test is reeeeeaaally not important enough to justify a lot of workarounds to get it passing.

Comment (dharders) [inline tests/integration/test.prefix.js]: No probs, just thought I'd share since you're now in the (lovely?) world of win dev ;)

**Comment by daleharvey** [comment]
Awesome, 👍

**Pull request merged**